app/controllers/work_move_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# a controller for changing the collection of work.
+class WorkMoveController < ApplicationController
+  before_action :authenticate_user!
+  verify_authorized except: %i[search]
+
+  def search
+    druid = params[:druid]
+    druid = "druid:#{druid}" unless druid.start_with?('druid:')
+    work = Work.find(params[:id])
+    collections = Collection.where(druid:)
+
+    render json: collections.map { |collection| collection_json(collection, work) }
+  end
+
+  def edit
+    authorize! :work_owner
+  end
+
+  def update
+    authorize! :work_owner
+
+    work = Work.find(params[:id])
+    collection = Collection.find(params[:collection])
+
+    work.update!(collection:)
```

Comment:
Should we re-run `CheckMoveWorkService` here? Is would be possible something changes between when they search for the item and when they update it, right?

Reply:
Added.